filebeat/input/filestream/input_delete_integration_test.go

```diff
@@ -0,0 +1,711 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build integration
+
+package filestream
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/gofrs/uuid/v5"
+	"github.com/stretchr/testify/require"
+
+	loginp "github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile"
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	"github.com/elastic/beats/v7/libbeat/reader/readfile/encoding"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/monitoring"
+)
+
+func TestFilestreamDelete(t *testing.T) {
+	testCases := map[string]map[string]any{
+		"on EOF": {
+			"prospector.scanner.check_interval":     "1s",
+			"close.reader.on_eof":                   true,
+			"delete.enabled":                        true,
+			"prospector.scanner.fingerprint.length": 64,
+			"delete.grace_period":                   0,
+		},
+		"on Inactive": {
+			"prospector.scanner.check_interval":     "1s",
+			"close.on_state_change.inactive":        "1s",
```

Comment:
It'd be safer to explicitly set `close.reader.on_eof: false` (I know it is the default it just makes the test more explicitly)